(antu) add mibs and create ipsec job

csilva-cl opened the pull request:
- add xups mib to measure output load of ups
- add base network mibs to measure CPU, memory and sensors
- add arista ipsec mib 
- create arista ipsec job to scrape only hosts with "ipsec" in the hostname

jhoblitt reviewed (changes requested):
This looks plausible.

The `snmp.yml` build needs to be updated as well: https://github.com/lsst-it/k8s-cookbook/tree/master/rke2/ayekan/exporters/snmp-generator

csilva-cl commented:
> This looks plausible.
> 
> The `snmp.yml` build needs to be updated as well: https://github.com/lsst-it/k8s-cookbook/tree/master/rke2/ayekan/exporters/snmp-generator

That's correct, I plan to do that in a different PR, to not mix purposes.

csilva-cl commented:
I also removed several unnecessary mibs in the Makefile

jhoblitt commented on rke2/ayekan/exporters/snmp-generator/Makefile:
```diff
@@ -18,43 +18,28 @@ MIB_PATH := 'mibs'
 ARISTA_URL          := https://www.arista.com/assets/data/docs/MIBS
-CISCO_URL           := https://raw.githubusercontent.com/cisco/cisco-mibs/2d465cce2de4e67a3561d8e41e4c99b597558d4b/v2
+CISCO_URL           := https://raw.githubusercontent.com/cisco/cisco-mibs/05dbf50226f7df5f52fd2dd1a9c17759273fa0d0/v2/
+CISCO_URL2           := https://raw.githubusercontent.com/cisco/cisco-mibs/05dbf50226f7df5f52fd2dd1a9c17759273fa0d0/v1/
```
```suggestion
CISCO_URL2          := https://raw.githubusercontent.com/cisco/cisco-mibs/05dbf50226f7df5f52fd2dd1a9c17759273fa0d0/v1/
```

jhoblitt replied:
There's an extra space here.

Why do we need the old cisco MIB?

csilva-cl replied:
oh I didnt noticed the extra space., good catch.

The old cisco is needed for the CPU mibs. Newer mibs are based on processes and some of our equipment do not support it.

csilva-cl replied:
oh I didnt noticed the extra space. 

The old cisco is needed for the CPU mibs. Newer mibs are based on processes and some of our equipment do not support it.

jhoblitt commented on rke2/ayekan/exporters/snmp-generator/Makefile:
```diff
@@ -18,43 +18,28 @@ MIB_PATH := 'mibs'
-SERVERTECH_URL      := 'https://cdn10.servertech.com/assets/documents/documents/817/original/Sentry3.mib'
-SERVERTECH4_URL     := 'https://cdn10.servertech.com/assets/documents/documents/815/original/Sentry4.mib'
-SYNOLOGY_URL        := 'https://global.download.synology.com/download/Document/Software/DeveloperGuide/Firmware/DSM/All/enu/Synology_MIB_File.zip'
+NET_SNMP_URL        := https://raw.githubusercontent.com/net-snmp/net-snmp/refs/heads/master/mibs/
```
Can a tagged release be used?  This could result in builds of `snmp.yml` being non-reproducible.

jhoblitt commented on rke2/ayekan/exporters/snmp-generator/Makefile:
```diff
@@ -18,43 +18,28 @@ MIB_PATH := 'mibs'
 CURL_OPTS ?= -L -sS --retry 3 --retry-delay 3 --fail -A "Prometheus SNMP generator Makefile"
 
-APC_URL             := https://download.schneider-electric.com/files?p_enDocType=Firmware&p_File_Name=powernet451.mib&p_Doc_Ref=APC_POWERNETMIB_451_EN
+APC_URL             := https://download.schneider-electric.com/files?p_Doc_Ref=APC_POWERNETMIB_EN&p_enDocType=Firmware&p_File_Name=powernet457.mib
```
```suggestion
APC_URL             := https://download.schneider-electric.com/files?p_Doc_Ref=APC_POWERNETMIB_EN&p_enDocType=Firmware
```

jhoblitt replied:
We should try to match upstream here.

jhoblitt commented on rke2/ayekan/exporters/snmp-generator/Makefile:
```diff
@@ -18,43 +18,28 @@ MIB_PATH := 'mibs'
-READYNAS_URL        := https://www.downloads.netgear.com/files/ReadyNAS/READYNAS-MIB.txt
-READYDATAOS_URL     := https://www.circitor.fr/Mibs/Mib/R/READYDATAOS-MIB.mib
-XUPS_URL            := https://www.circitor.fr/Mibs/Mib/X/XUPS-MIB.mib
+XUPS_URL            := https://mibbrowser.online/mibs/XUPS-MIB.mib
```
Should we try to contribute this upstream?

csilva-cl replied:
I dont think that's the spirit of the upstream. Not all mibs are opensource and the maintainers have said that. There are literally thousands of mibs available, they only maintain a basic list in the makefile, so no, we shouldn't contribute to upstream since we could hit licensing issues.

csilva-cl commented:
I removed the Makefile from upstream. It is not manageable without pushing mibs. The licensing issues of some of them make it difficult to keep up with upstream.

I pushed all the mibs to a folder, so it's possible to reproduce the build of the snmp.yml. New mibs should be added to the mibs folder. I added a readme for explanation.

This is something we can improve, but for the time being, at least we can reproduce the snmp config without much effort.

jhoblitt reviewed (approved):
LGTM.
